devito/passes/clusters/cse.py

```diff
@@ -55,28 +88,29 @@ def _cse(maybe_exprs, make, min_cost=1, mode='default'):
     make : callable
         Build symbols to store temporary, redundant values.
     mode : str, optional
-        The CSE algorithm applied. Accepted: ['default'].
-    """
+        The CSE algorithm applied. Accepted: ['basic', 'smartsort', 'advanced'].
 
-    # Note: not defaulting to SymPy's CSE() function for three reasons:
-    # - it also captures array index access functions (eg, i+1 in A[i+1] and B[i+1]);
-    # - it sometimes "captures too much", losing factorization opportunities;
-    # - very slow
-    # TODO: a second "sympy" mode will be provided, relying on SymPy's CSE() but
-    # also ensuring some form of post-processing
-    assert mode == 'default'  # Only supported mode ATM
+    Notes
+    -----
+    We're not using SymPy's CSE for three reasons:
+
+        * It also captures array index access functions (e.g., i+1 in A[i+1]);
```

Comment:
In future, we will want to capture these for things like `x+1-ltkn` and `INT(abs(x))` etc, but we probably want some kind of filter specifying a minimum verbosity

Reply:
yeah

Reply:
We will also probably want a unique kind of variable name to improve readability. Something like `i0` rather than `r0`